modules/slo/main.tf

```diff
@@ -39,117 +36,25 @@ resource "local_file" "error_budget_policy" {
   filename = "${path.module}/code/error_budget_policy.json"
 }
 
-# Temporary code to replace module invocation (see bottom of this file) until
-# https://github.com/terraform-google-modules/terraform-google-event-function/issues/37
-# is fixed.
-resource "google_cloud_scheduler_job" "job" {
-  name        = local.full_name
-  project     = var.project_id
-  region      = var.region
-  description = var.config.slo_description
-  schedule    = var.schedule
-  time_zone   = var.time_zone
-
-  pubsub_target {
-    topic_name = "projects/${var.project_id}/topics/${module.pubsub_topic.topic}"
-    data       = var.message_data
-  }
-}
-
-module "pubsub_topic" {
-  source     = "terraform-google-modules/pubsub/google"
-  version    = "~> 1.0"
-  project_id = var.project_id
-  topic      = local.full_name
-}
-
-# Generate a random uuid that will regenerate when one of the file in the source
-# directory is updated.
-resource "random_uuid" "code_hash" {
-  keepers = {
-    for filename in fileset(local.function_source_directory, "**/*") :
-    filename => filemd5("${local.function_source_directory}/${filename}")
-  }
-}
-
-# Generate a random uuid that will regenerate when the SLO config or the Error
-# Budget Policy is updated.
-# Workaround for https://github.com/terraform-providers/terraform-provider-random/issues/95
-resource "random_uuid" "config_hash" {
-  keepers = {
-    for content in [local_file.slo.content, local_file.error_budget_policy.content] :
-    content => md5(content)
-  }
-}
-
-# Regenerate the archive whenever one of the Cloud Function code files, SLO
-# config or Error Budget policy is updated.
-data "archive_file" "main" {
-  type        = "zip"
-  output_path = pathexpand("code-${local.full_name}-${local.cf_suffix}.zip")
-  source_dir  = pathexpand(local.function_source_directory)
-  depends_on = [
-    local_file.slo,
-    local_file.error_budget_policy
-  ]
-}
-
-resource "google_storage_bucket" "main" {
-  name          = "${local.full_name}-${local.bucket_suffix}"
-  force_destroy = var.bucket_force_destroy
-  location      = var.region
-  project       = var.project_id
-  storage_class = "REGIONAL"
-  labels        = var.labels
-}
-
-resource "google_storage_bucket_object" "main" {
-  name                = "code-${local.full_name}-${local.cf_suffix}"
-  bucket              = google_storage_bucket.main.name
-  source              = data.archive_file.main.output_path
-  content_disposition = "attachment"
-  content_encoding    = "gzip"
-  content_type        = "application/zip"
+module "slo_cloud_function" {
+  # TODO update version once event-function is released with new functionality
+  # source                              = "github.com/terraform-google-modules/terraform-google-scheduled-function"
+  # version                             = "~> 1.1"
+  source                                = "github.com/taylorludwig/terraform-google-scheduled-function?ref=feature%2Fuse-new-event-function"
```

Comment:
don't forget to update this when https://github.com/terraform-google-modules/terraform-google-scheduled-function/pull/28 is merged and a new version has been deployed to Terraform hub.

Reply:
great thanks @ocervell. Updated both event and scheduled function to the new released versions that support the dependent file functionality